src/Components/Web.JS/src/Services/NavigationManager.ts

```diff
@@ -76,8 +76,18 @@ export function attachToEventDelegator(eventDelegator: EventDelegator): void {
     const anchorTarget = findAnchorTarget(event);
 
     if (anchorTarget && canProcessAnchor(anchorTarget)) {
-      const href = anchorTarget.getAttribute('href')!;
-      const absoluteHref = toAbsoluteUri(href);
+      let anchorHref = anchorTarget.getAttribute('href')!; 
+      if (anchorHref.startsWith('#')) {
+        anchorHref = location.hash.length > 1 ? location.href.replace(location.hash, anchorHref) : location.href + anchorHref;
```

Comment:
Rather than using `location.href.replace` and having two different cases for whether or not the URL already has a hash, could this be made more easy to follow with code like the following?

```
anchorHref = `${location.origin}${location.pathname}${location.search}${location.hash}`;
```

That way, the browser is dealing with most of the parsing details for you and you don't have to distinguish different cases.